oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/DelegatingOAuth2AuthorizedClientProvider.java

```diff
@@ -64,10 +64,17 @@ public DelegatingOAuth2AuthorizedClientProvider(List<OAuth2AuthorizedClientProvi
 	@Nullable
 	public OAuth2AuthorizedClient authorize(OAuth2AuthorizationContext context) {
 		Assert.notNull(context, "context cannot be null");
-		return this.authorizedClientProviders.stream()
-				.map(authorizedClientProvider -> authorizedClientProvider.authorize(context))
-				.filter(Objects::nonNull)
-				.findFirst()
-				.orElse(null);
+		final List<OAuth2AuthorizedClient> clients = new ArrayList<>();
```

Comment:
No need for a list. Just return the first non null value

Reply:
fix this.

Reply:
Let's remove the list completely here. Remove this line and change the for loop to return when it finds the first value.

Something like:

```java
for (OAuth2AuthorizedClientProvider authorizedClientProvider: authorizedClientProviders) {
    OAuth2AuthorizedClient oauth2AuthorizedClient = authorizedClientProvider.authorize(context);
    if (oauth2AuthorizedClient != null) {
        return oauth2AuthorizedClient;
    }
}
return null;
```